src/neo/Consensus/ConsensusService.cs

```diff
@@ -252,10 +281,79 @@ private void ExtendTimerByFactor(int maxDelayInBlockTimes)
                 ChangeTimer(nextDelay);
         }
 
-        private void OnConsensusPayload(ConsensusPayload payload)
+        private void TryToSaveFuturePayloads(ConsensusPayload payload)
+        {
+            // Limit the maximum future payload to be saved, 
+            // avoiding caching payload when node is lagged behind by more than furtherPayloadToSave locks
+            uint furtherPayloadToSave = 10;
+            if ((payload.BlockIndex >= context.Block.Index + 1) && (payload.BlockIndex <= context.Block.Index + furtherPayloadToSave))
+            {
+                ConsensusMessage futureMessage;
+                try
+                {
+                    futureMessage = payload.ConsensusMessage;
+                }
+                catch (FormatException)
```

Comment:
I prefer one generic catch

Reply:
@shargon, do you mean one generic for `OnConsensus` and here?

Reply:
More about only this.

```
catch
{
}
```

Because if we throw in the future a different kind, we have an error.

Reply:
@shargon, I will mark this as resolved because currently it is just a copy of the current try catch system.
Thus, I believe that another PR for solving this on the current example would be better than here.